jupyter_client/blocking/client.py

```diff
@@ -5,17 +5,75 @@
 # Copyright (c) Jupyter Development Team.
 # Distributed under the terms of the Modified BSD License.
 
+from __future__ import print_function
+
+from getpass import getpass
 try:
     from queue import Empty  # Python 3
 except ImportError:
     from Queue import Empty  # Python 2
+import sys
 import time
 
+import zmq
+
 from traitlets import Type
 from jupyter_client.channels import HBChannel
 from jupyter_client.client import KernelClient
 from .channels import ZMQSocketChannel
 
+try:
+    monotonic = time.monotonic
+except AttributeError:
+    # py2
+    monotonic = time.time # close enough
+
+try:
+    TimeoutError
+except NameError:
+    # py2
+    TimeoutError = RuntimeError
+
+try:
+    # py2
+    input = raw_input
+except NameError:
+    # py3
+    pass
+
+
+def reqrep(meth):
+    def wrapped(self, *args, **kwargs):
+        reply = kwargs.pop('reply', False)
+        timeout = kwargs.pop('timeout', None)
+        msg_id = meth(self, *args, **kwargs)
+        if not reply:
+            return msg_id
+
+        return self._recv_reply(msg_id, timeout=timeout)
+    
+    basedoc, _ = meth.__doc__.split('Returns\n', 1)
+    parts = [basedoc.strip()]
+    if 'Parameters' not in basedoc:
+        parts.append("""
+        Parameters
+        ----------
+        """)
+    parts.append("""
```

Comment:
Yeah, me, too. I couldn't think of a better way, though. If you have a less gross idea, I'm all for it.

The main reason I did this is that I can't see a good way to get a nice *signature* for the wrapped functions. If I had a good signature, I'd be happier with a simpler "See KernelClient.method for more details...". It's easy to inherit the wrapped method's signature exactly, but I didn't see a good way to do that and add arguments.